src/context.ts

```diff
@@ -11,12 +11,16 @@ export const Consumer = Context.Consumer;
 
 let hasWarnedAboutDefault = false;
 
-export const useClient = (): Client => {
-  const client = useContext(Context);
+export const useClient = (client?: Client): Client => {
+  const context = useContext(Context);
```

Comment:
I would personally prefer keeping this named client

Reply:
I'd vote against that as the internal variable name inside this function is less important than the name of the parameter.

Reply:
If you feel strongly about this though then I can rename it to e.g. "customClient".

Reply:
I’d say, either are not that important, but either aren’t really clear. I’d call the inner variable “client” and the argument “override” or “overrideClient” as it overrides the context’s one.

The usage is already clear since the hook itself is called “useClient”

Reply:
I still believe that "client" is the best name for the parameter but overrideClient works for me too.